File: app/src/main/java/in/woloo/www/home/adapter/WolooStoreImagesAdapter.java
package in.woloo.www.home.adapter;

import android.content.Context;
import android.text.TextUtils;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;
import com.bumptech.glide.load.resource.bitmap.CenterCrop;
import com.bumptech.glide.load.resource.bitmap.RoundedCorners;
import com.bumptech.glide.request.RequestOptions;

import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;
import in.woloo.www.R;
import in.woloo.www.common.CommonUtils;
import in.woloo.www.dashboard.ui.home.model.NearByStoreResponse;
import in.woloo.www.utils.ImageUtil;
import in.woloo.www.utils.Logger;

public class WolooStoreImagesAdapter extends RecyclerView.Adapter<WolooStoreImagesAdapter.ViewHolder> {

    private Context context;
    private List<String> dataList;

    public WolooStoreImagesAdapter(Context context,List<String> dataList) {
        this.context = context;
        this.dataList=dataList;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        LayoutInflater layoutInflater = LayoutInflater.from(parent.getContext());
        View listItem= layoutInflater.inflate(R.layout.item_store_info_marker, parent, false);
        ViewHolder viewHolder = new ViewHolder(listItem);
        return viewHolder;
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        try {
            Logger.i("onBindViewHolder",""+position);
            holder.setStoreData(dataList.get(position));
        } catch (Exception e) {
             CommonUtils.printStackTrace(e);
        }
    }

    @Override
    public int getItemCount() {
       /* if (dataList!=null && dataList.size()>0){
            return dataList.size();
        }*/
        return dataList.size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder {
        @BindView(R.id.image_store_offer)
        ImageView image_store_offer;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            ButterKnife.bind(this,itemView);
        }

      private void  setStoreData(String image){
          try {
              //ImageUtil.loadImage(context,image_store_offer,image);
              Glide.with(context)
                      .load(image)
                      .apply(new RequestOptions().transform(new CenterCrop(), new RoundedCorners(16)))
                      .placeholder(R.drawable.banner_logo)
                      .into(image_store_offer);
          } catch (Exception e) {
               CommonUtils.printStackTrace(e);
          }
      }
    }
}

File: app/src/main/java/in/woloo/www/application_kotlin/adapters/loo_discovery/WolooStoreImagesAdapter.kt
package `in`.woloo.www.application_kotlin.adapters.loo_discovery

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import androidx.recyclerview.widget.RecyclerView
import butterknife.BindView
import butterknife.ButterKnife
import com.bumptech.glide.Glide
import com.bumptech.glide.load.resource.bitmap.CenterCrop
import com.bumptech.glide.load.resource.bitmap.RoundedCorners
import com.bumptech.glide.request.RequestOptions
import `in`.woloo.www.R
import `in`.woloo.www.common.CommonUtils
import `in`.woloo.www.utils.Logger

class WolooStoreImagesAdapter(private val context: Context, private val dataList: List<String>) :
    RecyclerView.Adapter<WolooStoreImagesAdapter.ViewHolder?>() {
    override fun onCreateViewHolder(
        parent: ViewGroup,
        viewType: Int
    ): ViewHolder {
        val layoutInflater = LayoutInflater.from(parent.context)
        val listItem: View =
            layoutInflater.inflate(R.layout.item_store_info_marker, parent, false)
        return ViewHolder(listItem)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        try {
            Logger.i("onBindViewHolder", "" + position)
            holder.setStoreData(dataList[position])
        } catch (e: Exception) {
            CommonUtils.printStackTrace(e)
        }
    }

    override fun getItemCount(): Int {
        /* if (dataList!=null && dataList.size()>0){
            return dataList.size();
        }*/
        return dataList.size
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        @JvmField
        @BindView(R.id.image_store_offer)
        var image_store_offer: ImageView? = null

        init {
            ButterKnife.bind(this, itemView)
        }

        fun setStoreData(image: String) {
            try {
                //ImageUtil.loadImage(context,image_store_offer,image);
                Glide.with(context)
                    .load(image)
                    .apply(RequestOptions().transform(CenterCrop(), RoundedCorners(16)))
                    .placeholder(R.drawable.banner_logo)
                    .into(image_store_offer!!)
            } catch (e: Exception) {
                CommonUtils.printStackTrace(e)
            }
        }
    }
}
